src/structures/ThreadChannel.js

```diff
@@ -0,0 +1,410 @@
+     */
+    this.archiveTimestamp = data.thread_metadata.archive_timestamp
+      ? new Date(data.thread_metadata.archive_timestamp).getTime()
+      : null;
+
+    /**
+     * The approximate count of messages in this thread
+     * <info> This value will not count above 50 even when there are more than 50 messages
+     * If you need an count higher than this, use ThreadChannel#messages.cache.size</info>
+     * @type {?number}
+     */
+    this.messageCount = data.message_count;
+
+    /**
+     * The approximate count of users in this thread
+     * <info> This value will not count above 50 even when there are more than 50 members</info>
+     * @type {?number}
+     */
+    this.memberCount = data.member_count;
+
+    if (data.member) this.members.add(data.member);
+    if (data.messages) for (const message of data.messages) this.messages.add(message);
+  }
+
+  /**
+   * A collection of the guild member objects for each of this threads members
+   * @type {Collection<Snowflake, GuildMember>}
+   * @readonly
+   */
+  get guildMembers() {
+    const members = new Collection();
+    this.members.cache.forEach(member => members.set(member.id, member.guildMember));
+    return members;
+  }
+
+  /**
+   * The time the thread was last archived or unarchived at
+   * @type {Date}
+   * @readonly
+   */
+  get archivedAt() {
+    return new Date(this.archiveTimestamp);
+  }
+
+  /**
+   * The parent channel of this thread
+   * @type {?NewsChannel|TextChannel}
+   * @readonly
+   */
+  get parent() {
+    return this.guild.channels.cache.get(this.parentID) ?? null;
+  }
+
+  /**
+   * Adds a member to this thread.
+   * @param {UserResolvable} member The member to add
+   * @param {string} [reason] The reason for adding this member
+   * @returns {Promise<Snowflake>}
+   */
+  addMember(member, reason) {
+    const id = this.client.users.resolveID(member);
+    if (!id) return Promise.reject(new TypeError('INVALID_TYPE', 'member', 'UserResolvable'));
+    return this.client.api
+      .channels(this.thread.id, 'thread-members', id)
+      .put({ reason })
+      .then(() => id);
+  }
+
+  /**
+   * Makes the client user join the thread.
+   * @returns {Promise<ThreadChannel>}
+   */
+  join() {
+    return this.client.api
+      .channels(this.id, 'thread-members', '@me')
+      .put()
+      .then(() => this);
+  }
+
+  /**
+   * Makes the client user leave the thread.
+   * @returns {Promise<ThreadChannel>}
+   */
+  leave() {
+    return this.client.api
+      .channels(this.id, 'thread-members', '@me')
+      .delete()
+      .then(() => this);
+  }
+
+  /**
+   * Gets the overall set of permissions for a member or role in this threads' parent, taking into account overwrites.
+   * @param {GuildMemberResolvable|RoleResolvable} memberOrRole The member or role to obtain the overall permissions for
+   * @returns {?Readonly<Permissions>}
+   */
+  permissionsFor(memberOrRole) {
+    return this.parent?.permissionsFor(memberOrRole);
+  }
```

Comment:
How are you taking overwrites into account if you're just checking for the parent permissions? Am I missing something?

Reply:
How will you take that exception into account then?

Reply:
that's what the other permission bits are for, also, `ThreadChannel#joinable` does the checks needed for sending messages if you really wanted to use it